scripts/developer_tools/k_work_on_pr.py

```
@@ -123,9 +105,11 @@ def checkout_pr_branch(pr: dict) -> None:
         except subprocess.CalledProcessError as exc:
             print(f"Failed to fetch {branch_name} from fork {fork_full_name}: {exc}", file=sys.stderr)
             sys.exit(1)
-        _checkout_local_branch(local_branch, remote_ref)
-        # The remote is only needed to fetch the fork's branch; drop it once checked out.
-        subprocess.run(["git", "remote", "remove", remote_name], capture_output=True, check=False)
+        try:
+            _checkout_local_branch(local_branch, remote_ref)
+        finally:
+            # The remote is temporary and must also be removed after a failed checkout.
+            subprocess.run(["git", "remote", "remove", remote_name], capture_output=True, check=False)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)